repo/grpc_repository_client.go

```diff
@@ -855,16 +862,17 @@ func openGRPCAPIRepository(ctx context.Context, si *APIServerInfo, password stri
 		return nil, errors.Wrap(err, "dial error")
 	}
 
-	par.refCountedCloser.registerEarlyCloseFunc(
-		func(ctx context.Context) error {
-			return errors.Wrap(conn.Close(), "error closing GRPC connection")
-		})
-
 	rep, err := newGRPCAPIRepositoryForConnection(ctx, conn, WriteSessionOptions{}, true, par)
 	if err != nil {
 		return nil, err
 	}
 
+	par.refCountedCloser.registerEarlyCloseFunc(
+		func(ctx context.Context) error {
+			rep.CloseDebug(ctx)
+			return errors.Wrap(conn.Close(), "error closing GRPC connection")
+		})
+
```

Comment:
`newGRPCAPIRepositoryForConnection()` is the who initializes the profile buffers internally, correct ?

If so, would it be better if `newGRPCAPIRepositoryForConnection` does the callback registration for stopping the buffers internally ?